started next auth

Tobbiloba opened the pull request.

michaelshimeles committed:
remove authorized in prod site

michaelshimeles committed:
smol bug

michaelshimeles committed:
Upgrading to nextjs 15

michaelshimeles committed:
Upgrading to nextjs 15

michaelshimeles committed:
Fix build

natesena committed:
first commit

natesena committed:
first commit

natesena committed:
Googlepixel (#2)

* first user created in db

* Working on Stripe

* User credits increasing

* Able to see list of customer transactions

* have transactions and credits pages

* have wavform audio

* Adjusted styling and have better dashboard, btu errors in building surrounding cookies

* Beginning design, no errors in build

* Many aesthetic improvements

* aesthetic2

* update import in root layout

* renamed navbar

* lots of debugging

* correct logs are happening in example webhooks

* halfway through switching to prisma code for backend

* working on making individual call data from vapi be env controllable

* Resizing of threejs scene working well

* fixed robots.ts and sitemap

* reviews looking good

* Pushing prisma changes to prod

* adjust secrets

* adjust gitignore to exlcude webhooks

* remove webhook files

* tested checkout completion success with stripe

* reindeer animation and different spacing on homepage

* a good way through the terms and conditions

* have terms + privacy page and db working. Need to integrate in checkout flow

* working on consent modal

* consent modal in shape

* consent modal done

* changed Call to call

* adjusted the link in the legal-page to be correct

* adding gocumentation to readme. Still need to fix deductions for webhooks which requires getting user id from the phone number of the call

* finished adding routes to readme

* getting errors trying to change the usercreate to add phone number

* added checkout flow to empty dashboards

* correctly debiting user credits

* trying to make the callsync only available on localhost

* account debiting is working

* fixed reindeer animation

* fixed color match of FAQ

* changed FAQ call recording, changed image links. Working on Imap Image.

* changed pricing to say 3x calls for 2 minutes

* testing imap image

* changed the url for images for reviews

* new url again for images

* update image domains in next config

* tried adding error message to vapi

* change error type

* working on getting new users in db

* change prod id + price id for 3 calls

* changed price to credits

* may have fixed user struggle with onboarding

* fixed bad characters

* added call credit back to credit page

* Add promo code

* change opengraph image url

* redeploying changes

* testing change for transactions query

* add support email plus gtag

* updating homepage

* added new CTA

* Added correct youtube video

* fixed video size

* removed music, added tiktok pixel

* returned the youtube and added google pixel

Tobbiloba committed:
started next auth

Tobbiloba committed:
completed next auth

digitaleydrive commented:
# **Authentication System Documentation**  
### with NextAuth, SMTP Email Verification, and Twilio Phone OTP Verification

## **Features**

---

### **1. User Authentication with NextAuth, Twilio, and Supabase**

The user authentication system utilizes **NextAuth**, **Twilio**, and **Supabase** to authenticate users and store additional user data. Here's a breakdown of the key components:

1. **NextAuth**: Manages user authentication and session management.
2. **Twilio**: Handles **SMS-based OTP verification** for user phone number verification.
3. **Supabase**: Stores additional user details in the user table and handles authentication.

---

### **2. Creating a User Table on Supabase**

To store user data beyond basic authentication, we created a **user table** in **Supabase**. The table includes the following columns:

- **id**: A unique identifier for each user (generated by Supabase).
- **first_name**: The user's first name.
- **password**: The user's password (Note: This is for internal use, not for direct authentication).
- **email**: The user's email address.
- **userImage**: URL for the user's profile image.
- **isPhoneVerified**: A boolean value indicating whether the user has verified their phone number.

This table ensures that all the additional data associated with a user can be stored and managed alongside their authentication details.

---

### **3. Middleware Component**

A **middleware** component was developed to handle user authentication and verification. The component ensures that users are authenticated before accessing protected routes or pages. Here's how it works:

- If a user is not authenticated (no active session), they are redirected to the **sign=in** page.
- If the user is authenticated, they are allowed to access protected routes.
- The component uses **NextAuth** to check the session status and determine if the user is logged in.

---

### **4. Creating a `useUser` Hook**

To simplify accessing user information across the application, a custom **`useUser` hook** was created. This hook retrieves the user's session and provides the following:

- **User information** (e.g., name, email, and profile data).
- A **`getUser`** function to refetch the user's data from **Supabase** or **NextAuth**.

---

### **5. Sign-In Page**

The **sign-in page** was updated from **Clerk** to a **custom-built solution** using **NextAuth** for authentication and **Supabase** for managing user authentication details. The process works as follows:

- **User Sign-Up**: When a user signs up, they are added to both:
  - **Supabase Authentication** for managing authentication.
  - **Supabase User Table** for storing additional information (e.g., first name, user image, etc.).
- **Redirection to Login**: After sign-up, the user is redirected to the **login screen**.
- **Sign in Page**: The user can log in using their **email** and **password**.
  - If the user has not verified their phone number, they will see a warning message asking them to verify it before proceeding.
  - The message will include a link to the **phone verification screen**.

---

### **6. Phone Number Verification**

To enhance security, the application includes a **phone number verification** process using **Twilio** for sending OTPs. Here's how it works:

- After user registration or login, if the user's phone number is not verified, a warning message is shown on the **login page**.
- The warning includes a link that redirects users to the **phone verification screen**.
- On the verification screen:
  - The system fetches the user's phone number from the **session**.
  - An **OTP** is automatically sent to the user's phone number via **Twilio**.
  - A **toast notification** is shown to the user informing them that the OTP has been sent.
  - The user enters the OTP they receive on their phone.
  - The OTP is verified against Twilio, and if it is correct, the user's phone number is marked as verified.
  - After successful verification, the user is redirected to the **home screen**, and the verification warning is removed.

---

### **7. Navbar with User Profile**

The **navbar** was designed to provide a seamless user experience depending on the user's authentication status. The following logic was implemented:

- **For unauthenticated users**:
  - The navbar displays **Login** and **Sign Up** buttons, allowing the user to authenticate or create a new account.
  
- **For authenticated users**:
  - The navbar shows the **user profile** with details like the user's profile picture and name, offering a personalized experience.
  - The user can click on the profile to access more options, such as **view profile**, **logout**, etc.

This ensures that users have an intuitive and dynamic navigation experience based on their authentication state.

---

### **8. Using Formik for Form Management**

For managing forms such as **sign-up**, **login**, and **profile updates**, **Formik** was integrated into the application. **Formik** simplifies form validation, state management, and submission, making it an excellent choice for handling user authentication forms.

- **Formik** handles form validation (with the help of **Yup**), ensuring that all fields are filled in properly and meet specific criteria.
- It also streamlines form handling, including:
  - Field-level validation
  - Handling form submission
  - Managing error messages


### **9. User Profile Page**

Work has started on the **user profile** page, which will allow authenticated users to view and edit their personal information. The profile page is designed to display essential user data, such as:

- **First name**
- **Email address**
- **Profile image**
- **Phone number** (and verification status)

At this stage, the profile page may be extended to include the following:

- **Password update** functionality.
- **Additional user details** (e.g., address, preferences).
- **Profile picture** update functionality.

While the page is in progress, the core features are being integrated, and additional user data will be added based on feedback and future requirements.


### **10. Conclusion**

The authentication system developed for this project successfully integrates **NextAuth**, **Twilio**, and **Supabase** to create a robust, scalable, and secure user authentication flow. Key features include:

- **Sign-up and sign in** with email/password authentication managed by **NextAuth**.
- **Phone number verification** through **Twilio** using OTPs.
- A **custom user table** in **Supabase** to store additional user data (e.g., first name, user image, and phone verification status).
- **Protected routing** to ensure that only authenticated users can access certain pages.
- **Form management** via **Formik** to simplify form handling and validation.

The system is designed to be flexible and easy to extend, with plans to add features such as **password recovery**, and an **enhanced user profile page** in future iterations. 

With these features, the authentication system provides a secure, efficient, and user-friendly experience for managing user accounts and data.

---

**End of Documentation.**







# **List of Modified Files and Descriptions**

This document provides a detailed overview of all files modified during the implementation of user authentication, phone number verification, and user profile management.

---

### **Modified Files**

- **app/(auth)/sign-in/[[...sign-in]]/page.tsx**  
  **Description**: Worked on the **sign-in form** and integrated **NextAuth** for handling user authentication. The form was customized to work with the authentication API and provide user feedback.

- **app/(auth)/sign-up/[[...sign-up]]/page.tsx**  
  **Description**: Developed the **sign-up form** to handle user registration. Integrated **Supabase Authentication** to manage user data and added logic for redirecting the user to the login page after sign-up.

- **app/(auth)/user-profile/[[...user-profile]]/page.tsx**  
  **Description**: Currently working on displaying the **user profile** page. The page fetches user data from the **Supabase user table** and displays it dynamically.

- **app/api/verify-phoneno/send-otp/route.ts**  
  **Description**: Created an **API endpoint** for sending and verifying OTPs for phone number verification using **Twilio**. This API allows users to verify their phone numbers by sending an OTP to their phone.

- **app/layout.tsx**  
  **Description**: Added **NextAuth** options to manage session storage and authentication state throughout the application.

- **app/page.tsx**  
  **Description**: Implemented a **phone number verification warning** on the home page, guiding users to verify their phone number if not already done.

- **app/provider.tsx**  
  **Description**: Removed the previous session provider and integrated the new **NextAuth** session provider to handle authentication state more efficiently.

- **app/verify-phoneno/page.tsx**  
  **Description**: Created the **phone number verification page** where users can enter the OTP received on their phone to complete the verification process.

- **components/ui/input.tsx**  
  **Description**: Modified the **input component** to handle various input types (email, password, OTP) and ensure validation is consistent across different forms.

- **components/user-profile.tsx**  
  **Description**: Started working on displaying user profile information, pulling data from **Supabase** to show user details such as name, email, and profile image.

- **components/wrapper/auth-wrapper.tsx**  
  **Description**: Created an **authentication wrapper** to protect routes that require user authentication. This ensures that unauthenticated users are redirected to the login page.

- **components/wrapper/navbar.tsx**  
  **Description**: Updated the **navbar** to dynamically change based on the user's authentication status. It shows either the **login** and **sign-up** buttons for unauthenticated users or the **user profile** for authenticated users.

- **hooks/use-user.ts**  
  **Description**: Created the **`useUser`** hook to fetch and store the user's authentication session. This hook allows easy access to user information and includes a function to refresh user data when needed.

- **lib/auth/login-credential.ts**  
  **Description**: Implemented logic for handling **login credentials**, including validating the provided credentials and integrating with **NextAuth** for authentication.

- **lib/auth/next-auth-options.ts**  
  **Description**: Defined the **NextAuth** configuration, including the authentication providers (such as **Supabase** and other login strategies) and session handling options.

- **next.config.js**  
  **Description**: Added **Supabase environment variables** for securely accessing the Supabase API and managing authentication within the app.

- **package-lock.json**  
  **Description**: Installed dependencies, including **Twilio** for OTP functionality, **Formik** for form handling, and other required libraries for user authentication and form validation.

- **package.json**  
  **Description**: Updated the **package.json** to include new dependencies for **Twilio**, **Formik**, and other necessary libraries required for phone number verification and form management.

- **utils/supabase.ts**  
  **Description**: Updated the **Supabase utility functions** to handle authentication, fetching user data, and performing operations related to user management (such as signing up, logging in, and updating user details).

---

![Screenshot from 2024-12-17 21-00-48](https://github.com/user-attachments/assets/0cd5191c-03b9-4037-a890-84ee7721fccc)
![Screenshot from 2024-12-17 21-01-11](https://github.com/user-attachments/assets/c17b2ef5-06ac-4b56-a1a5-94520acb4dd4)
![Screenshot from 2024-12-17 21-01-23](https://github.com/user-attachments/assets/3db12e51-b564-47c3-803b-5ba571ac9ca0)
![Screenshot from 2024-12-17 21-01-33](https://github.com/user-attachments/assets/45fb3434-616c-4663-bfe1-8f14d22ef7d9)

Tobbiloba commented:
# **Authentication System Documentation**  
### with NextAuth, SMTP Email Verification, and Twilio Phone OTP Verification

## **Features**

---

### **1. User Authentication with NextAuth, Twilio, and Supabase**

The user authentication system utilizes **NextAuth**, **Twilio**, and **Supabase** to authenticate users and store additional user data. Here's a breakdown of the key components:

1. **NextAuth**: Manages user authentication and session management.
2. **Twilio**: Handles **SMS-based OTP verification** for user phone number verification.
3. **Supabase**: Stores additional user details in the user table and handles authentication.

---

### **2. Creating a User Table on Supabase**

To store user data beyond basic authentication, we created a **user table** in **Supabase**. The table includes the following columns:

- **id**: A unique identifier for each user (generated by Supabase).
- **first_name**: The user's first name.
- **password**: The user's password (Note: This is for internal use, not for direct authentication).
- **email**: The user's email address.
- **userImage**: URL for the user's profile image.
- **isPhoneVerified**: A boolean value indicating whether the user has verified their phone number.

This table ensures that all the additional data associated with a user can be stored and managed alongside their authentication details.

---

### **3. Middleware Component**

A **middleware** component was developed to handle user authentication and verification. The component ensures that users are authenticated before accessing protected routes or pages. Here's how it works:

- If a user is not authenticated (no active session), they are redirected to the **sign=in** page.
- If the user is authenticated, they are allowed to access protected routes.
- The component uses **NextAuth** to check the session status and determine if the user is logged in.

---

### **4. Creating a `useUser` Hook**

To simplify accessing user information across the application, a custom **`useUser` hook** was created. This hook retrieves the user's session and provides the following:

- **User information** (e.g., name, email, and profile data).
- A **`getUser`** function to refetch the user's data from **Supabase** or **NextAuth**.

---

### **5. Sign-In Page**

The **sign-in page** was updated from **Clerk** to a **custom-built solution** using **NextAuth** for authentication and **Supabase** for managing user authentication details. The process works as follows:

- **User Sign-Up**: When a user signs up, they are added to both:
  - **Supabase Authentication** for managing authentication.
  - **Supabase User Table** for storing additional information (e.g., first name, user image, etc.).
- **Redirection to Login**: After sign-up, the user is redirected to the **login screen**.
- **Sign in Page**: The user can log in using their **email** and **password**.
  - If the user has not verified their phone number, they will see a warning message asking them to verify it before proceeding.
  - The message will include a link to the **phone verification screen**.

---

### **6. Phone Number Verification**

To enhance security, the application includes a **phone number verification** process using **Twilio** for sending OTPs. Here's how it works:

- After user registration or login, if the user's phone number is not verified, a warning message is shown on the **login page**.
- The warning includes a link that redirects users to the **phone verification screen**.
- On the verification screen:
  - The system fetches the user's phone number from the **session**.
  - An **OTP** is automatically sent to the user's phone number via **Twilio**.
  - A **toast notification** is shown to the user informing them that the OTP has been sent.
  - The user enters the OTP they receive on their phone.
  - The OTP is verified against Twilio, and if it is correct, the user's phone number is marked as verified.
  - After successful verification, the user is redirected to the **home screen**, and the verification warning is removed.

---

### **7. Navbar with User Profile**

The **navbar** was designed to provide a seamless user experience depending on the user's authentication status. The following logic was implemented:

- **For unauthenticated users**:
  - The navbar displays **Login** and **Sign Up** buttons, allowing the user to authenticate or create a new account.
  
- **For authenticated users**:
  - The navbar shows the **user profile** with details like the user's profile picture and name, offering a personalized experience.
  - The user can click on the profile to access more options, such as **view profile**, **logout**, etc.

This ensures that users have an intuitive and dynamic navigation experience based on their authentication state.

---

### **8. Using Formik for Form Management**

For managing forms such as **sign-up**, **login**, and **profile updates**, **Formik** was integrated into the application. **Formik** simplifies form validation, state management, and submission, making it an excellent choice for handling user authentication forms.

- **Formik** handles form validation (with the help of **Yup**), ensuring that all fields are filled in properly and meet specific criteria.
- It also streamlines form handling, including:
  - Field-level validation
  - Handling form submission
  - Managing error messages


### **9. User Profile Page**

Work has started on the **user profile** page, which will allow authenticated users to view and edit their personal information. The profile page is designed to display essential user data, such as:

- **First name**
- **Email address**
- **Profile image**
- **Phone number** (and verification status)

At this stage, the profile page may be extended to include the following:

- **Password update** functionality.
- **Additional user details** (e.g., address, preferences).
- **Profile picture** update functionality.

While the page is in progress, the core features are being integrated, and additional user data will be added based on feedback and future requirements.


### **10. Conclusion**

The authentication system developed for this project successfully integrates **NextAuth**, **Twilio**, and **Supabase** to create a robust, scalable, and secure user authentication flow. Key features include:

- **Sign-up and sign in** with email/password authentication managed by **NextAuth**.
- **Phone number verification** through **Twilio** using OTPs.
- A **custom user table** in **Supabase** to store additional user data (e.g., first name, user image, and phone verification status).
- **Protected routing** to ensure that only authenticated users can access certain pages.
- **Form management** via **Formik** to simplify form handling and validation.

The system is designed to be flexible and easy to extend, with plans to add features such as **password recovery**, and an **enhanced user profile page** in future iterations. 

With these features, the authentication system provides a secure, efficient, and user-friendly experience for managing user accounts and data.

---

**End of Documentation.**







# **List of Modified Files and Descriptions**

This document provides a detailed overview of all files modified during the implementation of user authentication, phone number verification, and user profile management.

---

### **Modified Files**

- **app/(auth)/sign-in/[[...sign-in]]/page.tsx**  
  **Description**: Worked on the **sign-in form** and integrated **NextAuth** for handling user authentication. The form was customized to work with the authentication API and provide user feedback.

- **app/(auth)/sign-up/[[...sign-up]]/page.tsx**  
  **Description**: Developed the **sign-up form** to handle user registration. Integrated **Supabase Authentication** to manage user data and added logic for redirecting the user to the login page after sign-up.

- **app/(auth)/user-profile/[[...user-profile]]/page.tsx**  
  **Description**: Currently working on displaying the **user profile** page. The page fetches user data from the **Supabase user table** and displays it dynamically.

- **app/api/verify-phoneno/send-otp/route.ts**  
  **Description**: Created an **API endpoint** for sending and verifying OTPs for phone number verification using **Twilio**. This API allows users to verify their phone numbers by sending an OTP to their phone.

- **app/layout.tsx**  
  **Description**: Added **NextAuth** options to manage session storage and authentication state throughout the application.

- **app/page.tsx**  
  **Description**: Implemented a **phone number verification warning** on the home page, guiding users to verify their phone number if not already done.

- **app/provider.tsx**  
  **Description**: Removed the previous session provider and integrated the new **NextAuth** session provider to handle authentication state more efficiently.

- **app/verify-phoneno/page.tsx**  
  **Description**: Created the **phone number verification page** where users can enter the OTP received on their phone to complete the verification process.

- **components/ui/input.tsx**  
  **Description**: Modified the **input component** to handle various input types (email, password, OTP) and ensure validation is consistent across different forms.

- **components/user-profile.tsx**  
  **Description**: Started working on displaying user profile information, pulling data from **Supabase** to show user details such as name, email, and profile image.

- **components/wrapper/auth-wrapper.tsx**  
  **Description**: Created an **authentication wrapper** to protect routes that require user authentication. This ensures that unauthenticated users are redirected to the login page.

- **components/wrapper/navbar.tsx**  
  **Description**: Updated the **navbar** to dynamically change based on the user's authentication status. It shows either the **login** and **sign-up** buttons for unauthenticated users or the **user profile** for authenticated users.

- **hooks/use-user.ts**  
  **Description**: Created the **`useUser`** hook to fetch and store the user's authentication session. This hook allows easy access to user information and includes a function to refresh user data when needed.

- **lib/auth/login-credential.ts**  
  **Description**: Implemented logic for handling **login credentials**, including validating the provided credentials and integrating with **NextAuth** for authentication.

- **lib/auth/next-auth-options.ts**  
  **Description**: Defined the **NextAuth** configuration, including the authentication providers (such as **Supabase** and other login strategies) and session handling options.

- **next.config.js**  
  **Description**: Added **Supabase environment variables** for securely accessing the Supabase API and managing authentication within the app.

- **package-lock.json**  
  **Description**: Installed dependencies, including **Twilio** for OTP functionality, **Formik** for form handling, and other required libraries for user authentication and form validation.

- **package.json**  
  **Description**: Updated the **package.json** to include new dependencies for **Twilio**, **Formik**, and other necessary libraries required for phone number verification and form management.

- **utils/supabase.ts**  
  **Description**: Updated the **Supabase utility functions** to handle authentication, fetching user data, and performing operations related to user management (such as signing up, logging in, and updating user details).

---

![Screenshot from 2024-12-17 21-00-48](https://github.com/user-attachments/assets/0cd5191c-03b9-4037-a890-84ee7721fccc)
![Screenshot from 2024-12-17 21-01-11](https://github.com/user-attachments/assets/c17b2ef5-06ac-4b56-a1a5-94520acb4dd4)
![Screenshot from 2024-12-17 21-01-23](https://github.com/user-attachments/assets/3db12e51-b564-47c3-803b-5ba571ac9ca0)
![Screenshot from 2024-12-17 21-01-33](https://github.com/user-attachments/assets/45fb3434-616c-4663-bfe1-8f14d22ef7d9)

natesena commented on middleware.ts:
```diff
@@ -1,65 +1,84 @@
-  : () => false;
+export const PROTECTED_SUB_ROUTES = [
+    '/dashboard',
+    '/api/calls',
```
I would leave the * as there are nested routes

natesena commented on middleware.ts:
```diff
@@ -1,65 +1,84 @@
+export const PROTECTED_SUB_ROUTES = [
+    '/dashboard',
+    '/api/calls',
+    '/api/user',
```
same here

natesena commented on .env.template:
```diff
@@ -22,4 +22,6 @@ STRIPE_SECRET_KEY=
-NEXT_PUBLIC_STRIPE_PRICE_ID=
\ No newline at end of file
+NEXT_PUBLIC_STRIPE_PRICE_ID=
+SUPABASE_URL=https://ztefoxytlvztnlgctbub.supabase.co
```
I would remove the keys from the git history entirely

natesena commented on .env.template:
```diff
@@ -22,4 +22,6 @@ STRIPE_SECRET_KEY=
\ No newline at end of file
+NEXT_PUBLIC_STRIPE_PRICE_ID=
+SUPABASE_URL=https://ztefoxytlvztnlgctbub.supabase.co
+SUPABASE_SERVICE_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inp0ZWZveHl0bHZ6dG5sZ2N0YnViIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTczMzc3OTU1OCwiZXhwIjoyMDQ5MzU1NTU4fQ.GE7P_uBy8sV4TIqZoJ_Mv0SfLVZknSxRpKDJkYtbWy4
```
same here

natesena commented on .env.test:
```diff
@@ -0,0 +1,28 @@
+NEXT_PUBLIC_CLERK_AFTER_SIGN_IN_URL=/
+NEXT_PUBLIC_CLERK_AFTER_SIGN_UP_URL=/
+
+SUPABASE_URL=https://ztefoxytlvztnlgctbub.supabase.co
```
lets just share keys a different way and not commit them

natesena commented on app/(auth)/sign-up/[[...sign-up]]/page.tsx:
```diff
@@ -1,24 +1,169 @@
+import Image from "next/image";
 import { useRouter } from "next/navigation";
+import { useState } from "react";
+import * as Yup from 'yup';
```
I am not familiar with this library. I'll check it out

natesena commented on app/(marketing)/marketing-page/page.tsx:
```diff
@@ -0,0 +1,96 @@
+import { Button } from '@/components/ui/button'
```
I dont need this page

natesena commented on app/dashboard/settings/page.tsx:
```diff
@@ -0,0 +1,39 @@
+"use client"
+import { Input } from '@/components/ui/input'
```
I deleted this page because the user cant edit anything.
Maybe in the future we open this page up, but I think it may confuse the user at this point.

natesena commented on app/page.tsx:
```diff
@@ -19,6 +20,7 @@ export default async function Home() {
             <SnowScene />
           </div>
           <div className="home-page-section relative z-10 min-h-[calc(100vh-var(--navbar-height))]">
+            <PhoneVerificationWarning />
```
Should this be conditionally rendered?

natesena commented on components/homepage/animated-logo-cloud.tsx:
```diff
@@ -0,0 +1,68 @@
+
```
We can delete this page

natesena commented on components/homepage/pricing.tsx:
```diff
@@ -0,0 +1,209 @@
+"use client";
```
I think I have this page elsewhere in my app. So I think we can remove it. Not certain though so we should double check.

natesena commented on components/homepage/static-logo-cloud.tsx:
```diff
@@ -0,0 +1,50 @@
+const logos = [
```
We can remove this I wont use it

natesena commented on components/wrapper/auth-wrapper.tsx:
```diff
@@ -1,17 +1,25 @@
-import config from '@/config';
+"use client";
+import { ReactNode } from "react";
+import { ClerkProvider } from "@clerk/nextjs";
```
Need to remove clerk here

natesena commented on components/wrapper/navbar.tsx:
```diff
@@ -23,13 +24,18 @@ const components: { title: string; href: string; description: string }[] = [
-  if (config?.auth?.enabled) {
-    const user = useAuth();
-    userId = user?.userId;
+  // if (config?.auth?.enabled) {
```
maybe can remove this comment

natesena commented on hooks/use-user.ts:
```diff
@@ -0,0 +1,33 @@
+      return;
+    }
+
+    const { data, error } = await supabase.from("user").select("*");
```
Generally speaking I like to use my API routes to get information from the database. Where is this data coming from?

natesena commented on public/images/user-profile/banner.jpeg:
I like to put images on my CDN instead so that they do not make vercel charge me fort he data :)

natesena commented on public/images/user-profile/banner.jpg:
Same comment here about images but nice touch!

Tobbiloba committed:
worked on feedbacks

Tobbiloba committed:
removing env files
